src/Form/WebPageArchiveEditForm.php

```diff
@@ -31,11 +32,13 @@ class WebPageArchiveEditForm extends WebPageArchiveFormBase {
    *
    * @param \Drupal\Core\Entity\EntityStorageInterface $web_page_archive_storage
    *   The storage.
+   * @param \Drupal\web_page_archive\Parser\SitemapParser $sitemap_parser
+   *   The sitemap parser service.
    * @param \Drupal\web_page_archive\Plugin\CaptureUtilityManager $capture_utility_manager
    *   The capture utility manager service.
    */
-  public function __construct(EntityStorageInterface $web_page_archive_storage, CaptureUtilityManager $capture_utility_manager) {
-    parent::__construct($web_page_archive_storage);
+  public function __construct(EntityStorageInterface $web_page_archive_storage, SitemapParser $sitemap_parser, CaptureUtilityManager $capture_utility_manager) {
```

Comment:
How about making the `SitemapParser` the third parameter with a null default? Sitemap usage is optional and making it a required param within method parameters seems heavy. The same would apply to `WebPageArchiveFormBase` but the parameter is already at the end, making it easy to convert to an optional parameter.

Reply:
I can do so, but it is worth noting that the `create()` method below this always injects that service in anyway. So I'm not entirely sure we're buying much in terms of performance here?

Reply:
Not concerned about performance, just method signatures. Adding parameters into the middle tends to reflect an API break, and form classes extending this (although there really aren't many) would be impacted. Fair?

Reply:
Fair enough.